### PR TITLE
QE: Add small checker to CVE audit scenario

### PR DESCRIPTION
**Pull request by nodeg**
## What does this PR change?

This will add a small check if the required patch is found in the step where the patch should be applied before selecting it.

                                           
                                                                                       
## GUI diff                         
                                                                                       
No difference.                           
                                           
- [x] **DONE**                        
                                                                                                                                                                              
## Documentation
- No documentation needed: only internal and user invisible changes


- [x] **DONE**

## Test coverage

- Cucumber tests were edited

- [x] **DONE**

## Links

- fixes https://github.com/SUSE/spacewalk/issues/20190
- Manager 4.3: https://github.com/SUSE/spacewalk/pull/20196
- Manager 4.2: https://github.com/SUSE/spacewalk/pull/20197
- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uy
uni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)


## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

                                                                                       
- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql" 
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"                                                                                                                               
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by elariekerboull — APPROVED** [review]
Could it be a product issue that the patch takes a long time to appear? 
Or is it definitely a testsuite issue?

**Review by Bischoff — APPROVED** [review]

**Review by srbarrios — APPROVED** [review]

**Comment by nodeg** [comment]
@elariekerboull: As mentioned in the linked issue, when running the same scenario manually on the controller and also doing all of it manually in the webUI, I ran into no issue. So FMPOV it is only a test suite issue.

File: testsuite/features/secondary/min_cve_audit.feature [hunk 1]
```diff
@@ -89,6 +89,7 @@ Feature: CVE Audit on SLE Salt Minions
     Given I am on the Systems overview page of this "sle_minion"
     When I follow "Software" in the content area
     And I follow "Patches" in the content area
+    And I wait at most 60 seconds until I see "milkyway-dummy-2345" text
```
Comment (srbarrios) [inline testsuite/features/secondary/min_cve_audit.feature:92]: If you tested, LGTM.
If not, consider to included "...refreshing the page" appended.

Comment (nodeg) [reply testsuite/features/secondary/min_cve_audit.feature:92]: I did not test this.
You mean e.g. that step @srbarrios?

```cucumber
And I wait until I see "milkyway-dummy-2345" text, refreshing the page
```

Comment (nodeg) [reply testsuite/features/secondary/min_cve_audit.feature:92]: Found it. I will adjust the PR.

Comment (nodeg) [reply testsuite/features/secondary/min_cve_audit.feature:92]: I cannot find that step.

File: testsuite/features/step_definitions/navigation_steps.rb [hunk 2]
```diff
@@ -78,6 +78,20 @@
...
+    begin
+      accept_prompt do
+        execute_script 'window.location.reload()'
+      end
+    rescue Capybara::ModalNotFound
+      # ignored
+    end
+  end
+end
+
```
Comment (nodeg) [inline testsuite/features/step_definitions/navigation_steps.rb:94]: I basically copied the step directly above and adjusted it.

Comment (srbarrios) [reply testsuite/features/step_definitions/navigation_steps.rb:94]: Use this one please:
`When(/^I wait until I see "([^"]*)" (text|regex), refreshing the page$/) do |text, type|`

Comment (srbarrios) [reply testsuite/features/step_definitions/navigation_steps.rb]: Another option, refactor the current step:
```
When(/^I wait at most "([^"]*)" seconds until I (see|do not see) "([^"]*)" text$/) do |seconds, action, text|
  is_shown = action == 'see'
  next if has_no_text?(text, wait: 3) == is_shown
  repeat_until_timeout(message: "I still see text '#{text}'", timeout: seconds.to_i) do
    break if has_no_text?(text, wait: 3) == is_shown
  end
end
```

Comment (nodeg) [reply testsuite/features/step_definitions/navigation_steps.rb]: Ah, ok that uses the default timeout. Adjusted. Thank you!

**Review by srbarrios — APPROVED** [review]

**Pull request merged**